### ISSUE TITLE
Partial metadata update - what do you think?

### ISSUE THREAD

**Issue by Tbaut** [paritytech/parity-signer#1048]
Updating metadata takes 5 to 10min on older phones. Repeating this every now and then for several networks is time consuming.
I wanted to hear your opinions about partial metadata upgrade. Metadata don't change that much between versions. The time spent to update metadata could be drastically reduced. 

It may be interesting to:
- get the version of the metadata by scanning a QR code from Signer
- the online part computes the difference between 2 metadata versions (most recent vs what's on Signer)
- define a protocol to transmit the diff along with the info to upgrade to the latest metadata

Another concern I have is that I think that rpc nodes don't have the older metadata (let me know if I'm mistaking here) so that the online part would have to have a DB of older metadata versions in order to be able to do the diff.

Let me know your thoughts and concerns

**Comment by maflohr** [comment]
Had experimented with some older Android devices and came to the conclusion that the current FPS setting of 1/30 (FPS_NOM/FPS_DEN) is highly incompatible with older smartphones. Some devices need at least an FPS setting of 2/30. With suboptimal displays or display settings an even slower refresh rate is necessary, factors are contrast, hue and even refresh rate.

For a good support of older devices I recommend a FPS setting of 4/30. This increases the update speed significantly.

**Comment by Slesarew** [comment]
Wow, can you share what models are you experimenting with? I honestly though that my Xiaomi Mi1 with cracked camera is quite on the bottom (and I ended up using it for personal production Signer).

This issue will have very low priority, unfortunately, since I already kind of push older devices support further. It kind of does not make sense to store petty cash in Signer, and cheapest android phone with proper hardware strongbox (full support thus) is already around $100 on ebay - I suppose it's Pixel 3 now.

On a different note, before we start doing complex things to metadata (which will bring about a mess about metadata signing and handling errors resulting in more phishing scenarios), we should do the simplest thing - just compress it. It's a long, human-readable thing, mostly consisting of spaces and brackets, and the task kind of trivial. We'll have to keep legacy support though and differentiate between Signer versions on metadata portals, maybe something like "speed download" button like on old warez websites lol.

**Comment by maflohr** [comment]
My primary Signer device is a Xiaomi Redmi 9C, low budget phone from 2020 - primarily because of the huge battery life of 30 days.

Tested also with Lenovo Chromebook 10.1, Pixel 3, 4a, 5 and 6.

The Xiaomi 9C and the Lenovo Chromebook had issues scanning the animated QRs and both would benefit huge from a decreased framerates.

All the Pixels worked very well, can recommend the Pixel 3 too.

Your arguments of compression > partial updates make sense to me.

**Comment by montekki** [comment]
Can you give #1054 a try? Really wondering if the `mlkit` docs are right and it would improve things

**Comment by maflohr** [comment]
Tried 3 different devices, 2 different monitors with a lot of samples. #1054 makes no difference.
